### DISCUSSION TITLE
for larger projects, or only single files?

### DISCUSSION THREAD

**Discussion by grguthrie** [jplag/JPlag#523] [Q&A] [answered]
I have a set of submissions which all contain multiple files.
Sometimes students have renamed files in their submission, but I want to be able to submit a bunch of submitted directories with multiple files in each, and check for copied code between them.
Is this possible?

**Comment by sebinside** [comment, marked as answer]
Considering multiple files (and plagiarism between them) is not only possible but the default. Have a look at [this](https://github.com/jplag/JPlag/tree/master/jplag/src/test/resources/de/jplag/samples/PartialPlagiarism) example with five submissions, consisting of multiple files, each.

**Reply by grguthrie** [reply]
Thank your; your example, is a code repository?

**Reply by tsaglam** [reply]
The example is code from our test cases. For your course, just put all files of a submission in their own folder. The names of the files/classes do not matter to JPlag. Your folder should look like this:

```
/path/to/root-directory
├── Submission-1
│   ├── Main.java
│   └── util
│       └── Utils.java
├── ...
└── Submission-n
    ├── Main.java
    └── util
        └── Utils.java
```
For info see [our wiki](https://github.com/jplag/JPlag/wiki/1.-How-to-Use-JPlag).

**Comment by grguthrie** [comment]
Very good - works perfectly, very valuable tool.
Thanks.
